### PR TITLE
Improved support for APEv2 tags.

### PR DESCRIPTION
**Pull request by smithjd15**
Parse and write additional APEv2 fields.

### CODE REVIEW

**Comment by jonaski** [comment]
You need to add the filetype to song.h also, here:
https://github.com/clementine-player/Clementine/blob/master/src/core/song.h#L93

File: ext/libclementine-tagreader/tagreader.cpp [hunk 1]
```diff
@@ -1110,6 +1268,38 @@ QByteArray TagReader::LoadEmbeddedArt(const QString& filename) const {
...
+  // APE formats
+  auto apeTagCover = [&](TagLib::APE::Tag* tag) {
+    QByteArray cover;
+    const TagLib::APE::ItemListMap& items = tag->itemListMap();
+    TagLib::APE::ItemListMap::ConstIterator it =
+        items.find("COVER ART (FRONT)");
+    if (it != items.end()) {
+      TagLib::ByteVector data = it->second.binaryData();
+
+      int pos = data.find('\0') + 1;
```
Comment (hatstand) [inline ext/libclementine-tagreader/tagreader.cpp]: Can you bounds check this?

Comment (smithjd15) [reply ext/libclementine-tagreader/tagreader.cpp]: How?

Comment (hatstand) [reply ext/libclementine-tagreader/tagreader.cpp]: 1. `pos != -1`
1. `pos < data.size()`

**Comment by smithjd15** [comment]
The following issues were fixed with this pr:
#4543,#2364,#1866,#4097

**Pull request merged**